packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -340,10 +339,27 @@ class FlutterPlugin implements Plugin<Project> {
                         "packages", "flutter_tools", "gradle", "src", "main", "kotlin",
                         "dependency_version_checker.gradle.kts")
                 project.apply from: dependencyCheckerPluginPath
-            } catch (Exception ignored) {
+            } catch (Exception e) {
```

Comment:
Why can't we do something like the suggestion below this comment with multiple catch blocks? Is is a quirk of throwing an exception from kotlin to groovy? 
```suggestion
            } catch (DependencyValidationException e) {
              // Rethrow 
            } catch (Exception e) {
              // eat exception Do other code path. 
            }
```

Reply:
> Is is a quirk of throwing an exception from kotlin to groovy?

Yes, that is correct. The `DependencyValidationException` gets wrapped in a couple of `GradleException`s

Reply:
I think when the Gradle plugin is fully rewritten to Kotlin, then we'll no longer have to depend on such hacks, right?

Reply:
Correct. I also haven't looked too far into this, but I think when the full re-write happens it can also be done in actual Kotlin source, and not Kotlin script (I believe AGP is written this way https://android.googlesource.com/platform/tools/base/+/1839aa23b8dc562005e2f0f0cc8e8b4c5caa37d0/build-system/gradle-core/src/main/java/com/android/build/gradle/internal/utils/agpVersionChecker.kt#58)

Reply:
Yes, that's my understanding as well (Kotlin "proper" instead of kts). Sweet!